src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,187 @@
+package duke;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Ui {
+
+    private static final String BORDER = "==================================================";
+    private Scanner conversation;
+
+    public Ui() {
+        this.conversation = new Scanner(System.in);
+    }
+
+    /**
+     * Returns a String of the user's next input line.
+     * @return User's next input line.
+     */
+    public String getNextLine() {
+        return conversation.nextLine();
+    }
+
+    /**
+     * Prints a message telling the user a Task has been added.
+     * @param tasks TaskList the Task has been added to.
+     */
+    public void printAddTaskMessage(TaskList tasks) {
+        System.out.println(BORDER);
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks.getLastTask());
+        System.out.printf("Now you have %d tasks in the list.\n", tasks.getTaskCount());
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a message to say goodbye to the user.
+     */
+    public void printByeMessage() {
+        System.out.println(BORDER);
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a message telling the user a Task has been deleted.
+     * @param task Deleted Task.
+     * @param tasks TaskList the Task has been deleted from.
+     */
+    public void printDeleteTaskMessage(Task task, TaskList tasks) {
+        System.out.println(BORDER);
+        System.out.println("Nice! I've deleted this task:");
+        System.out.println(task);
+        System.out.printf("Now you have %d tasks in the list.\n", tasks.getTaskCount());
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Print a message telling the user a Task has been marked as done.
+     * @param task Task marked as done.
+     */
+    public void printDoneTaskMessage(Task task) {
+        System.out.println(BORDER);
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(task);
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a message for when the user inputs an invalid command.
+     */
+    public void printUnknownMessage() {
+        System.out.println(BORDER);
+        System.out.println ("Unknown command!");
+        System.out.println(BORDER);
+    }
+
+    /**
+     * Prints a list of all Tasks in the TaskList.
+     * @param tasks TaskList to be printed.
+     */
+    public void printList(TaskList tasks) {
+        System.out.println(BORDER);
+        if (tasks.getTaskCount() > 0) {
```

Comment:
Maybe consider [making the happy path prominent](https://nus-cs2113-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent) here?